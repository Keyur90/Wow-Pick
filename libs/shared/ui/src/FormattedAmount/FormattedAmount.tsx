import React from 'react';
import NumberFormat from 'react-number-format';

interface FormattedAmountPropTypes {
  amount: number;
}

const FormattedAmount: React.FC<FormattedAmountPropTypes> = ({ amount }) => (
  <NumberFormat
    value={amount}
    displayType={'text'}
    thousandSeparator={true}
    decimalScale={2}
    fixedDecimalScale={true}
    prefix={'$'}
  />
);

export { FormattedAmount };
