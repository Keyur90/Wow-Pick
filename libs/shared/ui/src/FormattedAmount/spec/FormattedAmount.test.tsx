import React from 'react';
import { render } from '@testing-library/react';
import { FormattedAmount } from '../FormattedAmount';

test('renders FormattedAmount', () => {
  const { container } = render(<FormattedAmount amount={12345.67} />);

  expect(container).toMatchSnapshot();
});
