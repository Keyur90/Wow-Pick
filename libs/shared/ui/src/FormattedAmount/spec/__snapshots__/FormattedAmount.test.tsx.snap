// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FormattedAmount 1`] = `
<div>
  <span>
    $12,345.67
  </span>
</div>
`;
